p192/src/ecdsa.rs

```diff
@@ -0,0 +1,116 @@
+//! Elliptic Curve Digital Signature Algorithm (ECDSA)
+//!
+//! This module contains support for computing and verifying ECDSA signatures.
+//! To use it, you will need to enable one of the two following Cargo features:
+//!
+//! - `ecdsa-core`: provides only the [`Signature`] type (which represents an
+//!   ECDSA/P-192 signature). Does not require the `arithmetic` feature. This is
+//!   useful for 3rd-party crates which wish to use the `Signature` type for
+//!   interoperability purposes (particularly in conjunction with the
+//!   [`signature::Signer`] trait. Example use cases for this include other
+//!   software implementations of ECDSA/P-192 and wrappers for cloud KMS
+//!   services or hardware devices (HSM or crypto hardware wallet).
+//! - `ecdsa`: provides `ecdsa-core` features plus the [`SigningKey`] and
+//!   [`VerifyingKey`] types which natively implement ECDSA/P-192 signing and
+//!   verification.
+//!
+//! ## Signing/Verification Example
+//!
+//! This example requires the `ecdsa` Cargo feature is enabled:
+//!
+//! ```
+//! # #[cfg(feature = "ecdsa")]
+//! # {
+//! use p192::ecdsa::{signature::Signer, Signature, SigningKey};
+//! use rand_core::OsRng; // requires 'getrandom' feature
+//!
+//! // Signing
+//! let signing_key = SigningKey::random(&mut OsRng); // Serialize with `::to_bytes()`
+//! let message = b"ECDSA proves knowledge of a secret number in the context of a single message";
+//! let signature: Signature = signing_key.sign(message);
+//!
+//! // Verification
+//! use p192::ecdsa::{signature::Verifier, VerifyingKey};
+//!
+//! let verifying_key = VerifyingKey::from(&signing_key); // Serialize with `::to_encoded_point()`
+//! assert!(verifying_key.verify(message, &signature).is_ok());
+//! # }
+//! ```
+
+pub use ecdsa_core::signature::{self, Error};
+#[cfg(feature = "ecdsa")]
+use {
+    crate::{AffinePoint, Scalar},
+    ecdsa_core::hazmat::{SignPrimitive, VerifyPrimitive},
+};
+
+use super::NistP192;
+
+/// ECDSA/P-192 signature (fixed-size)
+pub type Signature = ecdsa_core::Signature<NistP192>;
+
+/// ECDSA/P-192 signature (ASN.1 DER encoded)
+pub type DerSignature = ecdsa_core::der::Signature<NistP192>;
+
+/// ECDSA/P-192 signing key
+#[cfg(feature = "ecdsa")]
+pub type SigningKey = ecdsa_core::SigningKey<NistP192>;
+
+/// ECDSA/P-192 verification key (i.e. public key)
+#[cfg(feature = "ecdsa")]
+pub type VerifyingKey = ecdsa_core::VerifyingKey<NistP192>;
+
+#[cfg(feature = "ecdsa")]
+impl SignPrimitive<NistP192> for Scalar {}
```

Comment:
I'd prefer to leave the signing components out for now, per NIST's guidance

Reply:
Done